### PR TITLE
[1507] Add missing variables during execution tool execution

### LINKED ISSUE #1507: Missing variables during execution of a source reconnection tool from the View description
* [ x] **I have checked that this feature has not yet been suggested by someone else.**

### Expected behavior

When implementing a source reconnection tool of a feature base edge, we are missing a way to access the semantic target of the edge.
In a feature base edge the variable "edgeSemanticElement" match the semantic element of the source bu we have no way to retrieve the semantic target of the edge.

So I would propose to expose:

* otherEnd - Samein compatibility mode
* edgeView - Same as compatibility mode
* semanticOtherEnd - New
* editingContext - Like it is done in other tools (since we now have access to IObjectService in ALQ services)

### PR DESCRIPTION
**Pull request by adaussy**
Bug: https://github.com/eclipse-sirius/sirius-components/issues/1507
Signed-off-by: Arthur Daussy <arthur.daussy@obeo.fr>

# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [ ] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: CHANGELOG.adoc [hunk 1]
```diff
@@ -44,6 +44,7 @@
 - https://github.com/eclipse-sirius/sirius-components/issues/1428[#1428] [layout] Add support for List layout compartment. We rely on the layout strategy engine handler switch and the layout engine handler switch to dispatch to the correct behavior depending on how children should be laid out and the type of the node. This kind of architecture has already been implemented for model operation for the View DSL. Nothing has changed if nodes have to be laid out freely.
 - https://github.com/eclipse-sirius/sirius-components/issues/1439[#1439] [form] Add support for toolbar actions in Form/FormDescriptionEditor
 - https://github.com/eclipse-sirius/sirius-components/issues/1437[#1437] [form] Add support for styles preview in FormDescriptionEditor
+- https://github.com/eclipse-sirius/sirius-components/issues/1507[#1507] [diagram] Missing variables during execution of a source reconnection tool from the View description
```
Comment (pcdavid) [inline CHANGELOG.adoc]: Maybe mention which variables were added?

File: packages/compatibility/backend/sirius-components-compatibility/src/main/java/org/eclipse/sirius/components/compatibility/services/diagrams/CompatibilityReconnectionToolsExecutor.java [hunk 2]
```diff
@@ -184,16 +184,18 @@ private boolean canReconnect(VariableManager variableManager, ReconnectEdgeDescr
...
         variableManager.put(IInterpreterSiriusVariables.DIAGRAM, toolInterpreterData.getDiagramContext().getDiagram());
         variableManager.put(IInterpreterSiriusVariables.SOURCE, toolInterpreterData.getSemanticReconnectionSource());
         variableManager.put(IInterpreterSiriusVariables.SOURCE_VIEW, toolInterpreterData.getReconnectionSourceView());
         variableManager.put(IInterpreterSiriusVariables.TARGET, toolInterpreterData.getSemanticReconnectionTarget());
         variableManager.put(IInterpreterSiriusVariables.TARGET_VIEW, toolInterpreterData.getReconnectionTargetView());
         variableManager.put(IInterpreterSiriusVariables.ELEMENT, toolInterpreterData.getSemanticElement());
-        variableManager.put("otherEnd", toolInterpreterData.getOtherEdgeEnd()); //$NON-NLS-1$
+        variableManager.put("semanticOtherEnd", toolInterpreterData.getOtherEdgeEnd()); //$NON-NLS-1$
+        variableManager.put("otherEnd", toolInterpreterData.getSemanticOtherEdgeEnd()); //$NON-NLS-1$
         variableManager.put("edgeView", toolInterpreterData.getEdgeView()); //$NON-NLS-1$
```
Comment (pcdavid) [inline packages/compatibility/backend/sirius-components-compatibility/src/main/java/org/eclipse/sirius/components/compatibility/services/diagrams/CompatibilityReconnectionToolsExecutor.java:197]: All these names should probably be constants defined in `ReconnectionToolInterpreterData`.

**Review by pcdavid — DISMISSED** [review]

File: packages/compatibility/backend/sirius-components-compatibility/src/main/java/org/eclipse/sirius/components/compatibility/services/diagrams/CompatibilityReconnectionToolsExecutor.java (hunk 2 above)
Comment (gcoutable) [reply packages/compatibility/backend/sirius-components-compatibility/src/main/java/org/eclipse/sirius/components/compatibility/services/diagrams/CompatibilityReconnectionToolsExecutor.java:197]: Should we consider the variable name change from `otherEnd` to `semanticOtherEnd` an API break to mention in the changelog?

Comment (adaussy) [reply packages/compatibility/backend/sirius-components-compatibility/src/main/java/org/eclipse/sirius/components/compatibility/services/diagrams/CompatibilityReconnectionToolsExecutor.java:197]: I have discussed that with SBE and he was not sure where to put them. He told me that anyway the variables names were more stable than the constants and their locations.

**Review by sbegaudeau — APPROVED** [review]

**Pull request merged**